src/VisualStudio/Core/Def/Implementation/Experimentation/KeybindingResetDetector.cs

```diff
@@ -233,21 +246,44 @@ private ReSharperStatus IsReSharperEnabled()
             }
 
             var cmds = new OLECMD[1];
-            cmds[0].cmdID = SuspendId;
+            cmds[0].cmdID = ResumeId;
             cmds[0].cmdf = 0;
 
-            var hr = _oleCommandTarget.QueryStatus(ReSharperCommandGroup, (uint)cmds.Length, cmds, IntPtr.Zero);
-            if (ErrorHandler.Failed(hr))
+            for (var count = 0; count < 10; count++)
             {
-                // In the case of an error when attempting to get the status, pretend that ReSharper isn't enabled. We also
-                // shut down monitoring so we don't keep hitting this.
-                FatalError.ReportWithoutCrash(Marshal.GetExceptionForHR(hr));
-                Shutdown();
-                return ReSharperStatus.NotInstalledOrDisabled;
+                var hr = _oleCommandTarget.QueryStatus(ReSharperCommandGroup, (uint)cmds.Length, cmds, IntPtr.Zero);
```

Comment:
I would extract only this part out and make it to run on UI thread and leave everything in BG.